src/iuse.cpp

```diff
@@ -8249,6 +8249,25 @@ int iuse::cable_attach(player *p, item *it, bool, const tripoint& )
     return 0;
 }
 
+int iuse::maple_tap(player *p, item *it, bool, const tripoint& )
+{
+    tripoint target;
+    std::string spile_name = item( "maple_tree_spile", 0 ).tname( 1 );
+
+    const std::string query = string_format( _( "Choose surrounding maple tree to use %s on?"), spile_name.c_str() );
+    if( !choose_adjacent( query, target ) ) {
+        return 0;
+    }
+
+    if( g->m.ter_at( target ).id == "t_tree_maple" ) {
+        iexamine::tree_maple( *p, target );
```

Comment:
If you're going to invoke the iexamine anyway, with no changes, just remove the iuse, because it doesn't add anything. The iexamine handles everything already, iuse doesn't add any convenience on top of that - it will actually ask for the spile once again.

Worse - if the spile is removed by the iuse, the line at the bottom (`return it->charges`) will invoke unexpected behavior since the `it` may no longer be a valid pointer.